### DISCUSSION TITLE
opensergo 与envoy 是否重复了？

### DISCUSSION THREAD

**Discussion by su-yh** [opensergo/opensergo-specification#83] [General]
在k8s 环境中如果使用了envoy 的功能的话，它已经提供了熔断降级与限流功能。

那么opensergo 所提供的这些功能就与envoy 冲突了。那么opensergo 似乎没有啥优势了？

**Comment by jnan806** [comment]
其实，OpenSergo 和 Envoy 不是一个层面的。
- OpenSergo 是一个标准，它定义了微服务治理相关的一系列标准。它的标准中涵盖了你提到的`熔断降级`与`限流功能`
- Envoy 实现了 `熔断降级`与`限流功能`。我们后期的规划会有插件使 Istio 体系 支持 OpenSergo 标准，例如 Envoy 
- 同时，Sentinel 也是 OpenSergo 的一个实现，在 Sentinel 2.0 的规划中，默认支持 OpenSergo 的标准。

`OpenSergo 和 Istio Envoy 的关系`类似于 `OpenSergo和Sentinl之间的关系`，可参考 [OpenSergo 与 Sentinel 的关系](
https://opensergo.io/zh-cn/docs/what-is-opensergo/faq/#opensergo-%E4%B8%8E-sentinel-%E7%9A%84%E5%85%B3%E7%B3%BB)

**Reply by su-yh** [reply]
我看了一下，在k8s 环境中，OpenSergo 定义了CRD，以sentinel 为例的话，OpenSergo 的控制面是不是相当于sentinel 的dashboard 的相关功能了，同时还代替了配置数据源(如 nacos) 来做配置的同步功能。
也就是说sentinel-core 将做为数据面以及熔断降级与限流的逻辑处理，而OpenSergo 作为它的控制面，提供类似sentinel-dashboard 的界面操作能力？

这两者的关系跟Envoy 与istio 之间的关系是一样的？

**Reply by jnan806** [reply]
OpenSergo 控制面 作为服务端 通过与SDK之间的通讯，实现了与客户端的配置同步，在这一点上可以说OpenSergo控制面是数据面的数据源。

OpenSergo 控制面目前还没有界面操作能力，该界面能力后续会推出，至于此操作功能后续可能会集成opensego-dashboard中，具体方案还待定。

OpenSergo 在功能上来说，除了数据源，dashboard这两块之外，还带有一些其他的功能，例如：一些计算可以放在服务端，结果回传给客户端之类。这些功能会以插件的形式来进行支持，详细可以参考 [issue#46 控制面扩展机制](https://github.com/opensergo/opensergo-control-plane/issues/46)

Envoy 与 istio 之间是 数据面与控制面之间的关系
而Sentinel除了是OpenSergo的数据面之外，Sentinel同时也是OpenSergo标准的一种实现。

另外， Istio 和 Sentinel 后续同时支持 OpenSergo 标准之后，关于流量路由等功能，可以快速的从 Istio 切换的Sentinel。

**Reply by su-yh** [reply]
好的，谢谢。有点理解 了，但还有些抽象。
